### ISSUE TITLE
bug: esp32 diversity, allow antenna2

### ISSUE THREAD

**Issue by olliw42** [olliw42/mLRS#200]
on the esp32 receivers with diversity, which use single spi, selecting antenna2 will make it crash. This is attributed to how the spi and sx are initialized.
Needs to be corrected however.

**Comment by brad112358** [comment]
Is there any chance this is related to https://github.com/olliw42/mLRS/issues/245 ?

**Comment by olliw42** [comment]
this one here - if I'm not mistaken - is understood. If I recall it is because with only sx2 the spi is not initialized. We would need an underlying level of structure which does the spi init either case, but not twice. We don't have that, and a straigth fiorward apporach would look ugly, in terms of arhcitecture.

**Comment by jlpoltrack** [comment]
Yes, unrelated.  Further - antenna 2 is disabled for all of the ESP32 diversity hardware. https://github.com/olliw42/mLRS/pull/203
